models/articles.models.js

```diff
@@ -36,23 +36,36 @@ exports.selectArticles = (topic, sort_by = "created_at", order = "desc") => {
     return Promise.reject({ status: 400, msg: "Bad request" });
   }
 
-  const queryValues = [];
-  let queryStr = `SELECT 
-  articles.article_id, articles.author, articles.title, articles.topic, articles.created_at, articles.votes, articles.article_img_url, 
-  COUNT(comments.comment_id)::INT AS comment_count
-  FROM articles
-  LEFT JOIN comments ON articles.article_id = comments.article_id`;
-
   if (topic) {
-    queryValues.push(topic);
-    queryStr += ` WHERE topic = $1`;
+    return db
+      .query("SELECT DISTINCT topic FROM articles WHERE topic = $1", [topic])
+      .then(({ rows }) => {
+        if (rows.length === 0) {
+          return Promise.reject({ status: 404, msg: "Not found" });
+        }
+
+        const queryValues = [topic];
+        let queryStr = `SELECT 
+        articles.article_id, articles.author, articles.title, articles.topic, articles.created_at, articles.votes, articles.article_img_url, 
+        COUNT(comments.comment_id)::INT AS comment_count
+        FROM articles
+        LEFT JOIN comments ON articles.article_id = comments.article_id
+        WHERE topic = $1
+        GROUP BY articles.article_id
+        ORDER BY articles.${sort_by} ${order};`;
+
+        return db.query(queryStr, queryValues).then(({ rows }) => rows);
+      });
   }
+  const queryStr = `SELECT 
+      articles.article_id, articles.author, articles.title, articles.topic, articles.created_at, articles.votes, articles.article_img_url, 
+      COUNT(comments.comment_id)::INT AS comment_count
+      FROM articles
+      LEFT JOIN comments ON articles.article_id = comments.article_id
+      GROUP BY articles.article_id
+      ORDER BY articles.${sort_by} ${order};`;
 
-  queryStr += ` GROUP BY articles.article_id
-  ORDER BY articles.${sort_by} ${order};`;
-  return db.query(queryStr, queryValues).then(({ rows }) => {
-    return rows;
-  });
+  return db.query(queryStr).then(({ rows }) => rows);
 };
```

Comment:
Could this model be DRY'er? Could the queryString be built up if there is a topic instead of having two separate queries to the database depending on the input?